guides/docker_guidance.md

```diff
@@ -0,0 +1,217 @@
+
+`docker-compose up -f docker-compose.yaml -f docker-compose.test.yaml`  
+
+It is also recommended not to expose any ports through Docker Compose used in CI as they may conflict with other ports already in use in the build agent.  
+
+Further documentation on docker-compose can be found at https://docs.docker.com/compose/reference/overview/#specifying-multiple-compose-files.
+
+### Use projects to provide unique volumes and networks
+To avoid conflicts when running different permutations of docker files, projects should be specified to segregate the volumes and networks.
+
+This can be achieved with the `-p` switch when calling docker compose on the command line.
+
+ i.e. to start the service
+
+`docker-compose -p ffc-demo-service -f docker-compose.yaml up`
+
+and to run the tests
+
+`docker-compose -p ffc-demo-service-test -f docker-compose.yaml -f docker-compose.test.yaml up`
+
+### Use environment variables to guarantee unique projects and containers
+When running through CI, a combination of the `-p` switch and environment variables can be used to ensure each build and test has unique project and container names.  This will prevent conflicts with other build pipelines when using tools such as a single node Jenkins.
+
+For example using Jenkins, the following compose files can be started via:
+
+`docker-compose -p ffc-demo-service-$PR_NUMBER-$BUILD_NUMBER -f docker-compose.yaml up`
+
+and tested with
+
+`docker-compose -p ffc-demo-service-test-$PR_NUMBER-$BUILD_NUMBER -f docker-compose.yaml -f docker-compose.test.yaml up`
+
+using `PR_NUMBER` and `BUILD_NUMBER` environment variables to isolate build tasks.
+
+`docker-compose.yaml`
+
+```
+version: '3.4'
+services:
+  ffc-demo-service:
+    build: .
+    image: ffc-demo-service
+    container_name: ffc-demo-service-${PR_NUMBER}-${BUILD_NUMBER}
+volumes:
+  node_modules: {}
+
+```
+
+`docker-compose.test.yaml`
+```
+version: '3.4'
+services:
+  ffc-demo-service:
+    command: npm run test
+    container_name: ffc-demo-service-test-${PR_NUMBER}-${BUILD_NUMBER}
+```
+
+### Composing multiple repositories for local development
+For scenarios where multiple containers need to be created across multiple repositories, it might be advantageous to create a "development" repo.
+
+The development repository would:
+- clone all necessary repositories
+- builds images from Dockerfiles in each repository by referencing Docker Compose files in those repositories
+- run containers based on those images in a single Docker network by referencing Docker Compose files in those repositories
+- run single containers for any shared dependencies across repositories such as message queues or databases
+
+To facilitate this, each repository with a potentially shared dependency will need it's Docker Compose override files to be setup in such a way that dependency containers can be isolated.  This will allow those repository services to run both in isolation and as part of wider service depending on development needs.
```

Comment:
Typo
```suggestion
To facilitate this, each repository with a potentially shared dependency will need its Docker Compose override files to be setup in such a way that dependency containers can be isolated.  This will allow those repository services to run both in isolation and as part of wider service depending on development needs.
```